services/src/datasets/in_memory.rs

```diff
@@ -189,11 +193,61 @@ impl DatasetStore<SimpleSession> for HashMapDatasetDb {
             symbology: dataset.symbology,
             provenance: dataset.provenance,
         };
-        self.backend.write().await.datasets.push(d);
+
+        let mut backend = self.backend.write().await;
+
+        backend.datasets.push(d);
+
+        if session.is_admin() {
+            backend.admin_datasets.push(id);
+        }
 
         Ok(id)
     }
 
+    async fn delete_dataset(&self, session: &SimpleSession, dataset_id: DatasetId) -> Result<()> {
```

Comment:
Either is an actix type. I don't think we should use it in our own code. I did it this way in order to have the interface homogeneous. In Pro, we don't need the admin flag, because we have the system user and rights management.